lmfit/confidence.py

```diff
@@ -413,12 +413,6 @@ def conf_interval2d(minimizer, result, x_name, y_name, nx=10, ny=10,
     redchi = result.redchi
     org = copy_vals(result.params)
 
-    def chi2_compare(best, current):
-        return current.chisqr
-
-    if prob_func is None:
-        prob_func = chi2_compare if chi2_out else f_compare
```

Comment:
when removing `prob_func` from the function here (and below), then it shouldn't be needed anymore as  keyword argument in line 356 - correct?

Reply:
@reneeotten good point, I updated the docstring and added a deprecation warning.